pypxe/tftp.py

```diff
@@ -123,13 +129,14 @@ def read(self, address, message):
             self.logger.error("Mode '{mode}' not supported".format(mode = mode))
             self.tftpError(address, 5, 'Mode {mode} not supported'.format(mode = mode))
             return
-        filename = self.filename(message)
+        req_file = self.filename(message)
+        filename = os.path.abspath(self.netbootDirectory + os.sep + req_file)
```

Comment:
This line works if the `chroot` is unsuccessful, but not if successful in certain cases. Once we `chroot` into `self.netbootDirectory`, `self.netbootDirectory` becomes `/`. Therefore prefixing our path with it will try to use a non-existing directory in the request.